### DISCUSSION TITLE
Uisng the WPF UI Template I dont see how one adds child menu items?

### DISCUSSION THREAD

**Discussion by ghost** [lepoco/wpfui#713] [Q&A]
I want to be able to add sub menu items to the  model how does one do that  for example the sales would have child items of Orders , Invoice, Recept?

  NavigationItems = new ObservableCollection<INavigationControl>
  {
      new NavigationItem()
      {
          Content = "Home",
          PageTag = "dashboard",
          Icon = SymbolRegular.Home24,
          PageType = typeof(Views.Pages.DashboardPage)
      },    new NavigationItem()
      {
          Content = "Sales",
          PageTag = "sales",
          Icon = SymbolRegular.CurrencyDollarEuro24,
          PageType = typeof(Views.Pages.SalesOrderPage)
      },    new NavigationItem()
      {
          Content = "Invoice",
          PageTag = "invoice",
          Icon = SymbolRegular.Receipt24,
          PageType = typeof(Views.Pages.DashboardPage)
      },    new NavigationItem()
      {
          Content = "Drivers",
          PageTag = "drivers",
          Icon = SymbolRegular.Receipt24,
          PageType = typeof(Views.Pages.DashboardPage)
      },    new NavigationItem()
      {
          Content = "Routes",
          PageTag = "routes",
          Icon = SymbolRegular.Receipt24,
          PageType = typeof(Views.Pages.DashboardPage)
      },    new NavigationItem()
      {
          Content = "Drivers Progress",
          PageTag = "driversprogress",
          Icon = SymbolRegular.VehicleTruck24,
          PageType = typeof(Views.Pages.DashboardPage)
      },    new NavigationItem()
      {
          Content = "Warehouse",
          PageTag = "warehouse",
          Icon = SymbolRegular.VehicleTruckCube24,
          PageType = typeof(Views.Pages.DashboardPage)
      }
      ,    new NavigationItem()
      {
          Content = "Pick Lists",
          PageTag = "picklists",
          Icon = SymbolRegular.List24,
          PageType = typeof(Views.Pages.DashboardPage)
          
      },

      new NavigationItem()
      {
          Content = "Data",
          PageTag = "data",
          Icon = SymbolRegular.DataHistogram24,
          PageType = typeof(Views.Pages.DataPage)
      }
  };

**Comment by pomianowski** [comment]
Hey @thehockeylabmn , if you are using the latest preview, you can create a submenu with code like this:
```c#
NavigationItems = new ObservableCollection<object>
{
    new NavigationViewItem()
    {
        Content = "My main menu",
        Icon = new SymbolIcon { Symbol = SymbolRegular.Fluent24 },
        MenuItems = new object[]
        {
            new NavigationViewItem("Sub menu item 1", SymbolRegular.Fluent24, typeof(MySubMenuPage1)),
            new NavigationViewItem("Sub menu item 2", SymbolRegular.Fluent24, typeof(MySubMenuPage2))
        }
    }
};
```

See:
https://github.com/lepoco/wpfui/blob/main/src/Wpf.Ui.Gallery/ViewModels/Windows/MainWindowViewModel.cs

**Reply by davidbuckleyni** [reply]
Thank u for reply will give this a shot can you keep the icons open like how the settings work in windows 11 @pomianowski

**Reply by davidbuckleyni** [reply]
When I attempt this i get this 

![image](https://github.com/lepoco/wpfui/assets/5619229/83025af7-82ef-4f04-b741-be5f88055258)


` ApplicationTitle = "Cella - Warehouse Managment System V1.0";

 NavigationItems = new ObservableCollection<INavigationControl>
 {
     new NavigationItem()
     {
         Content = "Home",
         PageTag = "dashboard",
         Icon = SymbolRegular.Home24,
         PageType = typeof(Views.Pages.DashboardPage)
     },    new NavigationItem()
     {
         Content = "Sales",
         PageTag = "sales",
         Icon = SymbolRegular.CurrencyDollarEuro24,
         PageType = typeof(Views.Pages.SalesOrderPage),

         MenuItems = new object[]
         {
             new NavigationViewItem("Sub menu item 1", SymbolRegular.Fluent24, typeof(MySubMenuPage1)),
             new NavigationViewItem("Sub menu item 2", SymbolRegular.Fluent24, typeof(MySubMenuPage2))
         }
     },    new NavigationItem()
     {
         Content = "Invoice",
         PageTag = "invoice",
         Icon = SymbolRegular.Receipt24,
         PageType = typeof(Views.Pages.DashboardPage)
     },    new NavigationItem()
     {
         Content = "Drivers",
         PageTag = "drivers",
         Icon = SymbolRegular.Receipt24,
         PageType = typeof(Views.Pages.DashboardPage)
     },    new NavigationItem()
     {
         Content = "Routes",
         PageTag = "routes",
         Icon = SymbolRegular.Receipt24,
         PageType = typeof(Views.Pages.DashboardPage)
     },    new NavigationItem()
     {
         Content = "Drivers Progress",
         PageTag = "driversprogress",
         Icon = SymbolRegular.VehicleTruck24,
         PageType = typeof(Views.Pages.DashboardPage)
     },    new NavigationItem()
     {
         Content = "Warehouse",
         PageTag = "warehouse",
         Icon = SymbolRegular.VehicleTruckCube24,
         PageType = typeof(Views.Pages.DashboardPage)
     }
     ,    new NavigationItem()
     {
         Content = "Pick Lists",
         PageTag = "picklists",
         Icon = SymbolRegular.List24,
         PageType = typeof(Views.Pages.DashboardPage)
         
     },

     new NavigationItem()
     {
         Content = "Data",
         PageTag = "data",
         Icon = SymbolRegular.DataHistogram24,
         PageType = typeof(Views.Pages.DataPage)
     },


 };`

**Comment by BoyFaceGirl** [comment]
> 
![image](https://github.com/lepoco/wpfui/assets/8773178/23003640-efa3-4699-897c-432e5fd00ceb)


![image](https://github.com/lepoco/wpfui/assets/8773178/9e6fa887-d8ac-4530-8a9f-41fb4b0c2f3c)

**Reply by davidbuckleyni** [reply]
This is not what i asked for what other person shown was enough thanks anyway

**Comment by davidbuckleyni** [comment]
@pomianowski im on 2.0.3 what is the lattest preview and how does one get it thanks
